dashboards_bundlers/local_deploy/jinja_templates/index.html.tpl

```diff
@@ -70,28 +67,26 @@
     <div id="dashboard-container" class="container" data-dashboard-layout="{{dashboardLayout}}">
         {%- for cell in nb.cells -%}
             {%- if (cell.metadata is defined) and (cell.metadata.urth is defined) and (cell.metadata.urth.dashboard is defined) -%}
+                {# Use cell layout metadata if available #}
                 {%- set hidden = cell.metadata.urth.dashboard.hidden -%}
                 {%- set layout = cell.metadata.urth.dashboard.layout -%}
+            {%- elif not hasNotebookMetadata -%}
+                {# If no notebook metadata available, we're showing everything in a default report mode #}
+                {%- set hidden = false -%}
+            {%- else -%}
+                {# If the notebook has dashboard metadata, but the cell is missing metadata, hide the cell because it's never been seen in 
+                   the dashboard layout mode. #}
+                {%- set hidden = true -%}
+                {%- set layout = none -%}
             {%- endif -%}
-            {%- if (dashboardLayout == "grid") and (layout == none) -%}
-                {%- if (useFallbackValues) -%}
-                    {%- set hidden = false -%}
-                    {%- set layout = { 'width': maxColumns, 'height': cellRowSpan, 'row': 0, 'col': 0 } -%}
-                {%- else -%}
-                    {%- set hidden = none -%}
-                    {%- set layout = none -%}
-                {%- endif -%}
-            {%- endif -%}
+            
             {# Thebe doesn't wrap HTML-converted cells with same structure as seen in Notebook,
                resulting in different styling. Adding necessary classes to match styles. #}
             {%- set textClass = "rendered_html" if (cell.cell_type is defined) and (cell.cell_type == "markdown") else "" -%}
             <div {% if (dashboardLayout == "report") -%}
-                    class="report-cell {{textClass}} {%- if (hidden) -%} dashboard-hidden{%- endif -%}"
+                    class="report-cell {{textClass}} {%- if (hidden) -%} dashboard-hidden {%- endif -%}"
```

Comment:
![image](https://cloud.githubusercontent.com/assets/6362426/14836660/55efa00e-0bd7-11e6-9bc5-3ae5611d3928.png)
We've got a collapsed space between classes here causing hidden cells to be visible. It looks like this was an existing problem.